x/profiles/keeper/msg_server_chain_link.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+func (k msgServer) LinkChainAccount(goCtx context.Context, msg *types.MsgLinkChainAccount) (*types.LinkChainAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	srcAddrData, err := types.UnpackAddressData(k.cdc, msg.SourceAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := srcAddrData.Validate(); err != nil {
+		return nil, err
+	}
+
+	if err := msg.SourceProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
+
+	if err := msg.DestinationProof.Verify(k.cdc); err != nil {
+		return nil, err
+	}
```

Comment:
Maybe we can move this inside the `StoreChainLink` method? This would avoid us having to duplicate this code inside the `OnReceivPacket` method as well